File: 13-RomanToInteger/13-RomanToInteger.cpp
// Last updated: 11/7/2025, 2:49:38 PM
class Solution {
public:
    int romanToInt(string s) {
        char arr[7]={'I','V','X','L','C','D','M'};
        int arr2[7]={1,5,10,50,100,500,1000};
        int num=0;
        for(int i=0;i<s.length();i++){
            for(int j=0;j<7;j++){
                if(s[i]==arr[j]){
                    if((s[i]=='I') || (s[i]=='X') || (s[i]=='C')){
                        if(s[i+1]==arr[j+1]){
                            num=num-arr2[j];
                        }

                        else if(s[i+1]==arr[j+2]){
                            num=num-arr2[j];
                        }

                        else num+=arr2[j];    
                    }
                    else num+=arr2[j];
                
                }
            }
        }
        return num;
    }
};

File: 13-RomanToInteger/13-RomanToInteger.go
// Last updated: 11/7/2025, 2:49:34 PM
func romanToInt(s string) int {
	input := []byte{'I', 'V', 'X', 'L', 'C', 'D', 'M'}
	output := []int{1, 5, 10, 50, 100, 500, 1000}

	ans := 0
	for i := range s {
		rimlan := slices.Index(input, s[i])
        num := output[rimlan]
        if i+1 < len(s) && ((s[i] == 'I') && (s[i+1] == 'V' || s[i+1] == 'X')) {
            num = -output[rimlan]
        } else if i+1 < len(s) && ((s[i] == 'X') && (s[i+1] == 'L' || s[i+1] == 'C')) {
            num = -output[rimlan]
        } else if i+1 < len(s) && ((s[i] == 'C') && (s[i+1] == 'D' || s[i+1] == 'M')) {
            num = -output[rimlan]
        }
		ans += num
	}

	return ans
}